django/core/cache/backends/memcached.py

```diff
@@ -214,3 +214,37 @@ def close(self, **kwargs):
         # libmemcached manages its own connections. Don't call disconnect_all()
         # as it resets the failover state and creates unnecessary reconnects.
         pass
+
+
+class PyMemcacheCache(BaseMemcachedCache):
+    """An implementation of a cache binding using pymemcache."""
+
+    def __init__(self, server, params):
+        import pymemcache.serde
+        super().__init__(server, params, library=pymemcache, value_not_found_exception=KeyError)
+        self._class = self._lib.HashClient
+        self._options = {
+            'allow_unicode_keys': True,
+            'default_noreply': False,
+            'serde': pymemcache.serde.pickle_serde,
+            **self._options,
+        }
+
+    @property
+    def client_servers(self):
+        output = []
+        for server in self._servers:
+            if server.startswith('unix:'):
+                output.append(server[5:])
+            elif server.startswith('/'):
+                output.append(server)
+            else:
+                if ':' not in server or server.endswith(']'):
+                    host, port = server, 11211
+                else:
+                    host, port = server.rsplit(':', 1)
+                    port = int(port)
+                if host.startswith('['):
+                    host = host.strip('[]')
+                output.append((host, port))
+        return output
```

Comment:
We can remove this if we require that `pymemcache>=3.4.0` is used as I included this when handling the IPv6 support. (See [here](https://github.com/pinterest/pymemcache/commit/9551dfd08e29728b5c77530536c54ad45d75a60e#diff-0b9fbbb2ef39034eb12da8f606537a0fR112-R130).)

```suggestion
```